File: DSA/RATNA_KHANNA_590013141/DAY_19_QUESTION_3.c
#include <stdio.h>
int searchMatrix(int rows, int cols, int matrix[rows][cols], int target) 
{
    int row = 0, col = cols - 1;
    while (row < rows && col >= 0) 
    {
        if (matrix[row][col] == target)
            return 1;
        else if (matrix[row][col] > target)
            col--;
        else
            row++;
    }
    return 0;
}
int main() 
{
    int rows, cols, target;
    printf("Enter number of rows: ");
    scanf("%d", &rows);
    printf("Enter number of columns: ");
    scanf("%d", &cols);
    int matrix[rows][cols];
    printf("Enter the elements of the matrix:\n");
    for (int i = 0; i < rows; i++) 
    {
        for (int j = 0; j < cols; j++) 
        {
            scanf("%d", &matrix[i][j]);
        }
    }
    printf("Enter the target value: ");
    scanf("%d", &target);
    int found = searchMatrix(rows, cols, matrix, target);
    if (found)
        printf("Output: true\n");
    else
        printf("Output: false\n");
    return 0;
}

File: DSA/LakshaKundra_590013237/Day_19_Question_3.py
def search_matrix(matrix, target):
    if not matrix:
        return False
    rows, cols = len(matrix), len(matrix[0])
    row, col = 0, cols - 1
    while row < rows and col >= 0:
        if matrix[row][col] == target:
            return True
        elif matrix[row][col] > target:
            col -= 1
        else:
            row += 1
    return False

print(search_matrix([[1,4,7],[2,5,8],[3,6,9]], 5))
print(search_matrix([[1,4,7],[2,5,8],[3,6,9]], 13))
print(search_matrix([[1,3,5],[2,4,6],[7,8,9]], 6))
